cpp/src/arrow/io/file.h

```diff
@@ -54,11 +54,12 @@ class ARROW_EXPORT FileOutputStream : public OutputStream {
   /// \brief Open a file descriptor for writing.  The underlying file isn't
   /// truncated.
   /// \param[in] fd file descriptor
+  /// \param[in] reuse should we use the page cache for the write
```

Comment:
```suggestion
  /// \param[in] reuse if false, the written data will not be kept in the OS page cache.
  ///                  This will have some negative impact on write throughput but can be
  ///                  used during large writes to avoid evicting more important data from
  ///                  the page cache.
```